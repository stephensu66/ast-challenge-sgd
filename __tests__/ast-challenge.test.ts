import  generateAST  from '../src/index';
import { readFileSync } from 'fs';
import { resolve } from 'path';

describe('AST Challenge', () => {
    it('should generate code correctly for a single method', () => {
        const resultingCode = generateAST(
            'UsePoolsQuery',
            'usePools',
            'QueryPoolsRequest',
            'QueryPoolsResponse',
            'pools',
            'poolsQuery'
        );
        expect(resultingCode).toMatchSnapshot();
        });


    it('should generate code for all example methods', () => {
        const exampleMethods = JSON.parse(readFileSync(resolve(__dirname, '../example-methods.json'), 'utf-8'));
    
        Object.keys(exampleMethods).forEach((methodName) => {
            const method = exampleMethods[methodName];
            const resultingCode = generateAST(
            `Use${methodName}Query`,
            `use${methodName}`,
            method.requestType,
            method.responseType,
            methodName,
            `${methodName.toLowerCase()}Query`
            );
            expect(resultingCode).toMatchSnapshot();
        });
        });
});