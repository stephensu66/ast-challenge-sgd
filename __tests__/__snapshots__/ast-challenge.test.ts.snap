// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AST Challenge should generate code correctly for a single method 1`] = `
"export interface UsePoolsQuery {
  request: QueryPoolsRequest;
  options: ReactQueryParams;
}

const usePools = (request, options) => {
  return useQuery([\\"poolsQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.pools(request);
  }, options);
};"
`;

exports[`AST Challenge should generate code for all example methods 1`] = `
"export interface UsePoolsQuery {
  request: QueryPoolsRequest;
  options: ReactQueryParams;
}

const usePools = (request, options) => {
  return useQuery([\\"poolsQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.Pools(request);
  }, options);
};"
`;

exports[`AST Challenge should generate code for all example methods 2`] = `
"export interface UseNumPoolsQuery {
  request: QueryNumPoolsRequest;
  options: ReactQueryParams;
}

const useNumPools = (request, options) => {
  return useQuery([\\"numpoolsQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.NumPools(request);
  }, options);
};"
`;

exports[`AST Challenge should generate code for all example methods 3`] = `
"export interface UseTotalLiquidityQuery {
  request: QueryTotalLiquidityRequest;
  options: ReactQueryParams;
}

const useTotalLiquidity = (request, options) => {
  return useQuery([\\"totalliquidityQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.TotalLiquidity(request);
  }, options);
};"
`;

exports[`AST Challenge should generate code for all example methods 4`] = `
"export interface UsePoolQuery {
  request: QueryPoolRequest;
  options: ReactQueryParams;
}

const usePool = (request, options) => {
  return useQuery([\\"poolQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.Pool(request);
  }, options);
};"
`;

exports[`AST Challenge should generate code for all example methods 5`] = `
"export interface UsePoolParamsQuery {
  request: QueryPoolParamsRequest;
  options: ReactQueryParams;
}

const usePoolParams = (request, options) => {
  return useQuery([\\"poolparamsQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.PoolParams(request);
  }, options);
};"
`;

exports[`AST Challenge should generate code for all example methods 6`] = `
"export interface UseTotalPoolLiquidityQuery {
  request: QueryTotalPoolLiquidityRequest;
  options: ReactQueryParams;
}

const useTotalPoolLiquidity = (request, options) => {
  return useQuery([\\"totalpoolliquidityQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.TotalPoolLiquidity(request);
  }, options);
};"
`;

exports[`AST Challenge should generate code for all example methods 7`] = `
"export interface UseTotalSharesQuery {
  request: QueryTotalSharesRequest;
  options: ReactQueryParams;
}

const useTotalShares = (request, options) => {
  return useQuery([\\"totalsharesQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.TotalShares(request);
  }, options);
};"
`;

exports[`AST Challenge should generate code for all example methods 8`] = `
"export interface UseSpotPriceQuery {
  request: QuerySpotPriceRequest;
  options: ReactQueryParams;
}

const useSpotPrice = (request, options) => {
  return useQuery([\\"spotpriceQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.SpotPrice(request);
  }, options);
};"
`;

exports[`AST Challenge should generate code for all example methods 9`] = `
"export interface UseEstimateSwapExactAmountInQuery {
  request: QuerySwapExactAmountInRequest;
  options: ReactQueryParams;
}

const useEstimateSwapExactAmountIn = (request, options) => {
  return useQuery([\\"estimateswapexactamountinQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.EstimateSwapExactAmountIn(request);
  }, options);
};"
`;

exports[`AST Challenge should generate code for all example methods 10`] = `
"export interface UseEstimateSwapExactAmountOutQuery {
  request: QuerySwapExactAmountOutRequest;
  options: ReactQueryParams;
}

const useEstimateSwapExactAmountOut = (request, options) => {
  return useQuery([\\"estimateswapexactamountoutQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.EstimateSwapExactAmountOut(request);
  }, options);
};"
`;
